janusgraph-test/src/test/java/org/janusgraph/graphdb/transaction/StandardTransactionBuilderTest.java

```diff
@@ -87,4 +88,26 @@ public void testConsistencyChecks() {
         tx = (StandardJanusGraphTx) graph.buildTransaction().consistencyChecks(false).start();
         assertFalse(tx.getConfiguration().hasVerifyUniqueness());
     }
+
+    @Test
+    public void testSkipDBCacheRead() {
+        StandardJanusGraphTx tx = (StandardJanusGraphTx) graph.newTransaction();
+        BackendTransaction backendTransaction = tx.getTxHandle();
+        assertFalse(tx.getConfiguration().isSkipDBCacheRead());
+        assertTrue(backendTransaction.isCacheEnabled());
+        tx.rollback();
+
+        tx = (StandardJanusGraphTx) graph.buildTransaction().skipDBCacheRead(true).start();
+        backendTransaction = tx.getTxHandle();
+        assertTrue(tx.getConfiguration().isSkipDBCacheRead());
+        assertFalse(backendTransaction.isCacheEnabled());
```

Comment:
Here we check that the cache is disabled

Reply:
`ExpirationKCVSCache` has methods such as: `getSlice` and `getSliceNoCache`.
Basically, when the cache is disabled in the transaction then any time we are getting a slice we will call `getSliceNoCache` which reads directly from the storage backend. If cache is enabled then a transaction will call `getSlice`.

Reply:
Here are the places where we check if we want to execute `getSlice` or `getSliceNoCache` depending if `cacheEnabled` is `true` or `false` in the current transaction:

https://github.com/JanusGraph/janusgraph/blob/2d4bbf81e5712db222a5f7ce4231cff90c276ee1/janusgraph-core/src/main/java/org/janusgraph/diskstorage/BackendTransaction.java#L274

https://github.com/JanusGraph/janusgraph/blob/2d4bbf81e5712db222a5f7ce4231cff90c276ee1/janusgraph-core/src/main/java/org/janusgraph/diskstorage/BackendTransaction.java#L290

https://github.com/JanusGraph/janusgraph/blob/2d4bbf81e5712db222a5f7ce4231cff90c276ee1/janusgraph-core/src/main/java/org/janusgraph/diskstorage/BackendTransaction.java#L401

Thus we can enable / disable cache access not only for the whole transaction but actually for part of the transaction. That said, this PR adds possibility to enable / disable cache for the whole transaction only. If users want to enable / disable cache for current transaction multiple times then they need to use the approach I posted in the PR description: https://github.com/JanusGraph/janusgraph/pull/3189#issue-1354789937

Reply:
Got it, I missed the current usage of `cacheEnabled` in `BackendTransaction`. I thought this was a new variable introduced by your PR. Now it's clear to me how it works. Thanks!